### PR TITLE
Optimize filter description when use nullable type

### PR DESCRIPTION
**Pull request by liuneng1994**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Optimize filter performance under nullable type

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by liuneng1994** [comment]
In my use case, there is a performance improvement of up to 35%

**Comment by liuneng1994** [comment]
Based on the status of has_null, I will submit more PRs to optimize the performance of Nullable

**Comment by liuneng1994** [comment]
AST fuzzer (ubsan) bugs don't look related to my code

File: src/Columns/ColumnNullable.cpp [hunk 1]
```diff
@@ -181,6 +188,9 @@ void ColumnNullable::insertRangeFrom(const IColumn & src, size_t start, size_t l
     const ColumnNullable & nullable_col = assert_cast<const ColumnNullable &>(src);
     getNullMapColumn().insertRangeFrom(*nullable_col.null_map, start, length);
     getNestedColumn().insertRangeFrom(*nullable_col.nested_column, start, length);
+    const auto& src_null_map_data = nullable_col.getNullMapData();
+    has_null = hasNull();
+    has_null |= contain_byte(src_null_map_data.data() + start, length, 1);
```
Comment (lgbo-ustc) [inline src/Columns/ColumnNullable.cpp:193]: `has_null = has_null ? has_null : contain_byte(src_null_map_data.data() + start, length, 1)`

File: src/Columns/ColumnNullable.cpp [hunk 2]
```diff
@@ -27,6 +27,11 @@ namespace ErrorCodes
     extern const int NOT_IMPLEMENTED;
 }
 
+template <typename T>
+inline bool contain_byte(const T* __restrict data, const size_t length, const signed char byte)
```
Comment (zhanglistar) [inline src/Columns/ColumnNullable.cpp:31]: Maybe memoryIsZero has better performance, should check it. And later memoryIsZero will be replaced by more higher performance function.

Comment (zhanglistar) [reply src/Columns/ColumnNullable.cpp:31]: std::memchr is 1.8x faster than memoryIsZero, but is slower than is 40% isa-l isisal_zero_detect.

**Comment by devcrafter** [comment]
> AST fuzzer (ubsan) bugs don't look related to my code

#42691

**Comment by devcrafter** [comment]
2-3x slow-downs in `jit_sort` perf test for table with nullables (`jit_test_memory_nullable`) look suspicious. 
See 
https://s3.amazonaws.com/clickhouse-test-reports/45876/9a17e01ce6ed008e291a833b24557c0a8197bf98/performance_comparison_[4/4]/report.html
https://s3.amazonaws.com/clickhouse-test-reports/45876/9a17e01ce6ed008e291a833b24557c0a8197bf98/performance_comparison_aarch64_[4/4]/report.html

@kitaisreal please could you have a look?

**Comment by liuneng1994** [comment]
@devcrafter The issue of jit_sort being slow seems to have been fixed

**Comment by liuneng1994** [comment]
After PR https://github.com/ClickHouse/ClickHouse/pull/45962 is entered, the optimization effect is not significant, but hasNull is an important interface, maybe I need to add other optimizations to prove the importance of this interface

**Comment by devcrafter** [comment]
@liuneng1994 I've figured out that we had similar PR before. Please see the comment which prevented merging it - https://github.com/ClickHouse/ClickHouse/pull/43598#issuecomment-1326316838.

**Pull request closed without merging**